upcloud/error.go

```diff
@@ -3,35 +3,151 @@ package upcloud
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud/client"
 )
 
+func NewError(clientErr *client.Error) error {
```

Comment:
This kinda has to stay exported because it has to be used in `service` package. We can't define this function in `service` package because it needs to set private fields (like `clientError`).

The alternative is just to move this whole logic back to `parseJSONServiceError` function, it's not that much logic anyway